_posts/2019-10-29-sacred-geometry.md

```diff
@@ -0,0 +1,428 @@
+---
+title:  "Javascript Animation"
+date: 2019-10-29 00:00:00 +0800
+author: jimmy
+---
+
+<div style="width: 85px; float: left; margin-right: 20px;">
+    <img src="/assets/images/articles/tt_small.svg">
+</div>
+
+A quick look at drawing and animation with HTML Canvas and Javascript
+<!--more-->
+
+**TL;DR** Check out this cool javascript animation of multiplication tables <a href='#final_canvas'>below.</a>
+
+My very first experience with programming was on Khan Academy, going through their Drawing with Java tutorial. I hated
+it. Programmatically placing circles and coloring them in, trying to make a penguin was frustrating beyond belief.
+Trying to guess where the centers of my circles needed to be so that my penguin's head sat _on_ its body, instead of
+down in its stomach like some avian _blemmyae_ eventually drove me into resignation, not to pick up programming again
+for a long time.
+
+## Basic drawing
+So how is it I sit here writing an article about going back to programmatic art? To put it simply, my understandings of
+what "turtle path" art is good for has evolved. A lot. One of the things that was so frustrating about trying to draw
+that penguin so many years ago was that I had no idea why I needed to be drawing it programmatically. I could click and
+draw this thing and be done so much faster. (Granted, it would still look like a second grader who just discovered MS
+Paint.) But the turtle paths I'll be explaining below I definitively could _not_ click and draw myself. This art is
+derived from mathematics (please, don't leave! It's simple math, I promise!) and as such, is suited perfectly for
+telling a turtle where to start, where to end, and what color to draw.
+
+With that said, before we look at the math, and the code to draw all of it, let's first look at some simple canvas drawing. We'll start with a line.
```

Comment:
may be annoying/too much effort to do, but having a "code pen" with this ready to experiment on could be nice?

Reply:
I created a github repo with the raw times tables animation code in it and linked below. Never really setup a code pen before...